### ISSUE TITLE
(touts): This duplicates some of the computations in softmax_op 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#2337]
because we need the intermediate (logits -max(logits)) values to
avoid a log(exp()) in the computation of the loss.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [8a59748c087a2fee535c0d5067dbabb01920e812](https://github.com/imdone/tensorflow/commit/8a59748c087a2fee535c0d5067dbabb01920e812) by A. Unique TensorFlower nobody@tensorflow.org*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/core/kernels/sparse_xent_op.h#L157-L164
